src/wallet/wallet.cpp

```diff
@@ -3958,11 +3947,13 @@ CWallet* CWallet::CreateWalletFromFile(const std::string walletFile)
             if (!walletInstance->SetHDMasterKey(masterPubKey))
                 throw std::runtime_error(std::string(__func__) + ": Storing master key failed");
         }
-        CPubKey newDefaultKey;
-        if (walletInstance->GetKeyFromPool(newDefaultKey, false)) {
-            walletInstance->SetDefaultKey(newDefaultKey);
-            if (!walletInstance->SetAddressBook(walletInstance->vchDefaultKey.GetID(), "", "receive")) {
-                InitError(_("Cannot write default address") += "\n");
+
+        // Generate an address for the address book and to fill the keypool
+        CPubKey key;
+        if (walletInstance->GetKeyFromPool(key, false)) {
```

Comment:
Can you call `walletInstance->TopUpKeyPool();` here instead of this entire section? No need to waste a key and an address book entry for this.